### ISSUE TITLE
Proposal: use `combobox` role

### ISSUE THREAD

**Issue by MatthewCharlton** [JedWatson/react-select#4695]
Currently `react-select` does not work with JAWS primarily due to how JAWS interprets the HTML elements on the page and what attributes those elements have. In the case of `react-select` using an `input` with no other attributes JAWS sees no reason why a user would press the down arrow and intercepts the keyboard event that `react-select` uses to open the menu. 

I spiked adding `role="combobox"` to the `input` and it seems to fix all the issues we see with JAWS while also improving the experience for the other screen readers too.

It turns out you don’t NEED to use the `aria-activedecendent` attribute to manage selected options, you are allowed to handle focus states of options yourself, in this case `react-select` can continue to use `aria-live` to announce focused and selected options to the user while letting the screen reader know that they are actually on a “Select/Combobox” not just a text field. The screen reader gives better feedback when the user interacts with the component without interfering with the live region announcements.

I feel like the combobox version gives better, more contextual feedback when used in conjunction with the `aria-live` announcements than just the announcements alone.

I have a CodeSandbox demo here: https://hl5h2.csb.app/

Here is a demo of how react-select acts with VoiceOver, NVDA and JAWS currently vs using combobox:

https://user-images.githubusercontent.com/33848730/126096922-98d4faab-c189-4bf8-84ac-5207f142ad91.mp4

Linked issues: #2937 #4062

**Comment by JedWatson** [comment]
Implemented by #4697
